sys/net/transport_layer/ng_udp/ng_udp.c

```diff
@@ -72,7 +72,7 @@ static uint16_t _calc_csum(ng_pktsnip_t *hdr, ng_pktsnip_t *pseudo_hdr,
         payload = payload->next;
     }
     /* process applicable UDP header bytes */
-    csum = ng_inet_csum(csum, (uint8_t *)hdr->data, 6);
+    csum = ng_inet_csum(csum, (uint8_t *)hdr->data, sizeof(ng_udp_hdr_t));
```

Comment:
@sgso but it is in `ng_udp_hdr_build()`. We are wasting also instructions by comparing the result of this function with the (host-byteorder adjusted) checksum field, instead of just to zero.